src/toggle-button/internal.tsx

```diff
@@ -0,0 +1,53 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+import React from 'react';
+import clsx from 'clsx';
+
+import InternalButton from '../button/internal';
+import { fireNonCancelableEvent } from '../internal/events';
+import { ToggleButtonProps } from './interfaces';
+import { getToggleIcon } from './util';
+
+import styles from './styles.css.js';
+
+export const InternalToggleButton = React.forwardRef(
+  (
+    {
+      pressed,
+      iconName: defaultIconName,
+      pressedIconName,
+      iconSvg: defaultIconSvg,
+      pressedIconSvg,
+      iconUrl: defaultIconUrl,
+      pressedIconUrl,
+      variant,
+      onChange,
+      className,
+      onClick,
+      ...rest
+    }: ToggleButtonProps,
+    ref: React.Ref<ToggleButtonProps.Ref>
+  ) => {
+    return (
+      <InternalButton
+        className={clsx(className, styles[`variant-${variant}`], { [styles.pressed]: pressed })}
+        variant={variant}
+        iconName={getToggleIcon(pressed, defaultIconName, pressedIconName)}
+        iconUrl={getToggleIcon(pressed, defaultIconUrl, pressedIconUrl)}
+        iconSvg={getToggleIcon(pressed, defaultIconSvg, pressedIconSvg)}
+        aria-pressed={pressed}
+        onClick={event => {
+          if (onClick) {
```

Comment:
But they can use `onChange` all the same, right?

Reply:
yes

Reply:
So why should `onClick` be necessary then?

Reply:
Good question. In this case they are interchangeable, so the `onClick` could be removed

Reply:
We can always add it later if we need it, but in case of doubt, it's usually preferable to lean towards not adding new props because removing them is harder than adding them (or impossible)

Reply:
updated